pkg/registry/core/service/storage/rest.go

```diff
@@ -521,8 +521,8 @@ func isValidAddress(ctx context.Context, addr *api.EndpointAddress, pods rest.Ge
 	if pod == nil {
 		return fmt.Errorf("pod is missing, skipping (%s/%s)", addr.TargetRef.Namespace, addr.TargetRef.Name)
 	}
-	if pod.Status.PodIP != addr.IP {
-		return fmt.Errorf("pod ip doesn't match endpoint ip, skipping: %s vs %s (%s/%s)", pod.Status.PodIP, addr.IP, addr.TargetRef.Namespace, addr.TargetRef.Name)
+	if pod.Status.PodIPs[0].IP != addr.IP {
```

Comment:
Do we need a sanity length check on PodIPs[] before referencing the [0] entry?

Reply:
yes, thanks